api/test/e2enew/migrate/migrate_test.go

```diff
@@ -0,0 +1,242 @@
+	"github.com/onsi/gomega"
+)
+
+const (
+	checksumLength = 4 // 4 bytes (uint32)
+)
+
+type AllData struct {
+	Counsumers    []interface{}
+	Routes        []interface{}
+	Services      []interface{}
+	SSLs          []interface{}
+	Upstreams     []interface{}
+	Scripts       []interface{}
+	GlobalPlugins []interface{}
+	PluginConfigs []interface{}
+}
+
+type response struct {
+	Code    int    `json:"Code"`
+	Message string `json:"Message"`
+	Data    struct {
+		ConflictItems AllData `json:"ConflictItems"`
+	} `json:"Data"`
+}
+
+var _ = ginkgo.Describe("Migrate", func() {
+	var exportData []byte
+
+	ginkgo.It("prepare config data", prepareConfigData)
+	ginkgo.It("export config success", func() {
+		req := base.ManagerApiExpect().GET("/apisix/admin/migrate/export")
+		resp := req.Expect()
+		resp.Status(http.StatusOK)
+		exportData = []byte(resp.Body().Raw())
+		data := exportData[:len(exportData)-checksumLength]
+		checksum := binary.BigEndian.Uint32(exportData[len(exportData)-checksumLength:])
+		gomega.Expect(checksum).Should(gomega.Equal(crc32.ChecksumIEEE(data)))
+	})
+
+	ginkgo.It("import config conflict and return", func() {
+		req := base.ManagerApiExpect().POST("/apisix/admin/migrate/import")
+		buffer := bytes.NewBuffer(exportData)
+		req.WithMultipart().WithForm(map[string]string{"mode": "return"})
+		req.WithMultipart().WithFile("file", "apisix-config.bak", buffer)
+		resp := req.Expect()
+		resp.Status(http.StatusOK)
+		rsp := &response{}
+		err := json.Unmarshal([]byte(resp.Body().Raw()), rsp)
+		gomega.Expect(err).Should(gomega.BeNil())
+		gomega.Expect(rsp.Code).Should(gomega.Equal(20001))
+		gomega.Expect(len(rsp.Data.ConflictItems.Routes)).Should(gomega.Equal(1))
+		gomega.Expect(len(rsp.Data.ConflictItems.Upstreams)).Should(gomega.Equal(1))
+		gomega.Expect(len(rsp.Data.ConflictItems.Services)).Should(gomega.Equal(1))
+	})
+
+	ginkgo.It("import config conflict and skip", func() {
+		req := base.ManagerApiExpect().POST("/apisix/admin/migrate/import")
+		buffer := bytes.NewBuffer(exportData)
+		req.WithMultipart().WithForm(map[string]string{"mode": "skip"})
+		req.WithMultipart().WithFile("file", "apisix-config.bak", buffer)
+		resp := req.Expect()
+		resp.Status(http.StatusOK)
+		rsp := &response{}
+		err := json.Unmarshal([]byte(resp.Body().Raw()), rsp)
+		gomega.Expect(err).Should(gomega.BeNil())
+		gomega.Expect(rsp.Code).Should(gomega.Equal(0))
+	})
+
+	ginkgo.It("import config conflict and overwrite", func() {
+		req := base.ManagerApiExpect().POST("/apisix/admin/migrate/import")
+		buffer := bytes.NewBuffer(exportData)
+		req.WithMultipart().WithForm(map[string]string{"mode": "overwrite"})
+		req.WithMultipart().WithFile("file", "apisix-config.bak", buffer)
+		resp := req.Expect()
+		resp.Status(http.StatusOK)
+		rsp := &response{}
+		err := json.Unmarshal([]byte(resp.Body().Raw()), rsp)
+		gomega.Expect(err).Should(gomega.BeNil())
+		gomega.Expect(rsp.Code).Should(gomega.Equal(0))
+	})
```

Comment:
Need some DP test after  imported.

For example, visit a certain route to ensure normal access.